noir-projects/noir-contracts/contracts/app/amm_contract/src/main.nr

```diff
@@ -98,14 +98,15 @@ pub contract AMM {
         let liquidity_token = Token::at(config.liquidity_token);
 
         let sender = context.msg_sender();
+        let this_address = context.this_address();
```

Comment:
`PrivateContext::this_address` does not invoke any oracles, it just returns a value in a struct